sky/server/server.py

```diff
@@ -649,6 +649,9 @@ async def dispatch(self, request: fastapi.Request, call_next):
 app.include_router(ssh_node_pools_rest.router,
                    prefix='/ssh_node_pools',
                    tags=['ssh_node_pools'])
+# increase the resource limit for the server
+soft, hard = resource.getrlimit(resource.RLIMIT_NOFILE)
+resource.setrlimit(resource.RLIMIT_NOFILE, (hard, hard))
```

Comment:
We have this warning in subprocess_utils.py: https://github.com/skypilot-org/skypilot/blob/0cbaa0afb1b47a748f7bd1aac7ed9e86e1e0e449/sky/utils/subprocess_utils.py#L68-L77 Should we consolidate the two somehow?